File: P1 2019.1/Questao(1)/python/entities/Produto.py
class Produto:
    def __init__(self, id_produto: str):
        self.id_produto = id_produto
        self.cor = str()
        self.preco = float()

    def __eq__(self, other):
        return [self.id_produto == other.id_produto if isinstance(other, Produto) else False]

    def __str__(self):
        return f"{self.id_produto} - {self.cor} {self.preco}"

File: P1 2019.1/Questao(1)/java/entities/Produto.java
package entities;

public class Produto {
    private String id;
    private String cor;
    private double preco;

    public Produto(String id) {
        this.id = id;
    }

    public String getId() {
        return id;
    }

    public String getCor() {
        return cor;
    }

    public void setCor(String cor) {
        this.cor = cor;
    }

    public double getPreco() {
        return preco;
    }

    public void setPreco(double preco) {
        this.preco = preco;
    }

    @Override
    public boolean equals(Object o) {
        if (o instanceof Produto) {
            return ((Produto) o).getId().compareTo(this.getId()) == 0;
        }
        return false;
    }

    @Override
    public String toString() {
        return "[" + getId()
                + " - "
                + getCor()
                + " - "
                + String.format("%.2f", getPreco())
                + "]";
    }
}
